params/config.go

```diff
@@ -185,6 +187,8 @@ var (
 		ShanghaiTime:                  newUint64(0),
 		CancunTime:                    newUint64(0),
 		Enable4844:                    false,
+		// TODO(0xHansLee): Need to bake timestamp for Odyssey hard fork
```

Comment:
We may use [1243000](https://odyssey.storyscan.xyz/block/countdown/1243000) as the upgrade height (9:58PM PT 12/12).

Reply:
I would propose it to use block height to be consistent with CL side upgrade reference and we can keep relative order between EL and CL since our block time is not consistent. [This](https://github.com/piplabs/story-geth/commit/a54ef9e719c7429ce70921a2296a9157a7c323dd#diff-10ff00a15ef58da7485fedeca3906c73cb236fde4b143a9d61601c63cbf1ffe8R161) was how we did in iliad.

@edisonz0718 we can discuss here.

Reply:
I'm ok changing to block but it seems that there are many places that call this base gas calculation function and some places may not have block number available for passing in. But will leave it to @0xHansLee to check

Reply:
My only concern to use block height for hardfork is the consistency in geth side. 
For example, when you see [this function](https://github.com/piplabs/story-geth/pull/64/files#diff-e2021ee5515b2e4e1a9fa5adfc57b95578103c5639809132acddbafd00599a11R57), I added an argument of time to see if the hardfork is activated or not. If we use block height for hardfork for story-geth and other hardfork from geth (Ethereum mainnet) happens based on time, this function may have 2 args, time and height. 

So, if we can make the upgrade of EL and CL backward compatible, I think it would be better to use timestamp for hardfork as done in geth.

Reply:
I think geth also used block height in some previous hard fork upgrade? such as [this ](https://github.com/piplabs/story-geth/pull/64/files#diff-10ff00a15ef58da7485fedeca3906c73cb236fde4b143a9d61601c63cbf1ffe8R633)

I think block time for eth is quite consistent, but we are not. Therefore, using block time will create inconsistency between geth and story.

Reply:
OK, then I will change to use block height. I think it would be ok to pass block height instead of timestamp as both are from block header in most use cases.

Reply:
And will bake the block height for Theogony hardfork on Odyssey as 1243000 @limengformal

Reply:
Reflected in [521aa76](https://github.com/piplabs/story-geth/pull/64/commits/521aa768baeb27414c7ce64c07cf19611832aaff).